### PR TITLE
(BKR-113) Host Prebuilt commands for FreeBSD

### PR DESCRIPTION
**Pull request by petems**

### CODE REVIEW

**Comment by anodelman** [comment]
We'll look at this post https://github.com/puppetlabs/beaker/pull/764 merge.

File: lib/beaker/host_prebuilt_steps.rb [hunk 1]
```diff
@@ -289,7 +292,13 @@ def get_ip(host)
     # @param [Host] host the host to act upon
     # @param [String] etc_hosts The string to append to the /etc/hosts file
     def set_etc_hosts(host, etc_hosts)
-      host.exec(Command.new("echo '#{etc_hosts}' > /etc/hosts"))
+      if host['platform'] =~ /freebsd/
+        # FreeBSD gets weird about special characters, we have to go a little OTT here
+        freebsd_hosts = etc_hosts.gsub(/\t/,'\\t').gsub(/\n/,'\\n')
```
Comment (anodelman) [inline lib/beaker/host_prebuilt_steps.rb]: This substitution is happening in multiple places - let's roll that into the freebsd host helper method.

File: lib/beaker/host_prebuilt_steps.rb [hunk 2]
```diff
@@ -289,7 +292,13 @@ def get_ip(host)
...
+        host.exec(Command.new("printf '#{freebsd_hosts}' > /etc/hosts"))
```
Comment (anodelman) [inline lib/beaker/host_prebuilt_steps.rb]: Might be good here to have a host method like host.echo_msg(msg), which can then be overridden for freebsd hosts.

File: lib/beaker/host_prebuilt_steps.rb (hunk 1 above)
Comment (petems) [inline lib/beaker/host_prebuilt_steps.rb]: Done :+1:

File: lib/beaker/host_prebuilt_steps.rb (hunk 2 above)
Comment (petems) [inline lib/beaker/host_prebuilt_steps.rb]: Done :+1:

**Comment by petems** [comment]
Done echo_to_file method for FreeBSD, need to add it as a default exec on Unix host, will do soon :+1:

**Pull request merged**